main.go

```diff
@@ -0,0 +1,310 @@
+			return fmt.Errorf("malformated line %q", line)
+		}
+
+		k, v = strings.TrimSpace(items[0]), strings.TrimSpace(items[1])
+		log.Debug("parsed configuration line",
+			"key", k,
+			"value", v)
+
+		switch k {
+		case "mysql_host":
+			mysqlHost = v
+
+		case "mysql_port":
+			mysqlPort = v
+
+		case "mysql_user":
+			mysqlUser = v
+
+		case "mysql_password":
+			mysqlPassword = v
+
+		case "mysql_database":
+			mysqlDatabase = v
+
+		case "riemann_host":
+			riemannHost = v
+
+		case "riemann_port":
+			riemannPort = v
+
+		case "interval":
+			i, err := strconv.ParseInt(v, 10, 32)
+			if err != nil {
+				return fmt.Errorf("invalid value %q for setting `interval`", v)
+			}
+			interval = time.Duration(i) * time.Second
+
+		case "delay":
+			d, err := strconv.ParseFloat(v, 32)
+			if err != nil {
+				return fmt.Errorf("invalid value %q for setting `delay`", v)
+			}
+			delay = d
+
+		case "hostname":
+			hostname = v
+
+		case "tags":
+			riemannTags = strings.Split(v, " ")
+
+		default:
+			log.Warn(fmt.Sprintf("unsupported configuration setting %q", k))
+		}
+	}
+	if err := scanner.Err(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func main() {
+	var (
+		riemann *raidman.Client
+		db      *mysql.Conn
+		err     error
+	)
+
+	// Handle termination signals
+	sig := make(chan os.Signal, 1)
+	stop := make(chan bool, 1)
+	signal.Notify(sig, syscall.SIGINT, syscall.SIGTERM)
+
+	go func() {
+		<-sig // Block until we receive a notification on the chan from signal handler
+		log.Info("terminating")
+		stop <- true
+	}()
+
+	if riemann, err = raidman.Dial("tcp4", net.JoinHostPort(riemannHost, riemannPort)); err != nil {
+		dieOnError(fmt.Sprintf("unable to connect to riemann server: %s", err))
+	}
+
+	go func() {
+		log.Info("starting")
+
+		for {
+			log.Debug("getting database handle")
+			if db, err = getDbHandle(db); err != nil {
+				log.Warn("unable to get database handle", "error", err)
+				time.Sleep(interval)
+				continue
+			}
+
+			events := make([]*raidman.Event, 0)
+			t := time.Now()
+
+			log.Debug("gathering statistics")
+			r, err := db.Execute("SHOW ALL SLAVES STATUS")
+			if err != nil {
+				log.Warn("unable to query replication status", "error", err)
+				events = append(events, &raidman.Event{
+					Time:        t.Unix(),
+					Service:     "mysql/replication",
+					State:       "unknown",
+					Description: fmt.Sprintf("unable to query replication status: %s", err),
+					Tags:        riemannTags,
+					Ttl:         float32(interval.Seconds() + delay),
+				})
+				goto send
+			}
+
+			for i := 0; i < r.Resultset.RowNumber(); i++ {
+				event := &raidman.Event{
+					Time:    t.Unix(),
+					Service: fmt.Sprintf("mysql/replication/conn%d", i),
+					State:   "ok",
+					Ttl:     float32(interval.Seconds() + delay),
+					Tags:    riemannTags,
+				}
+				if hostname != "" {
+					event.Host = hostname
+				}
+
+				if connName, _ := r.Resultset.GetStringByName(i, "Connection_name"); connName != "" {
```

Comment:
You should probably check the error here.

Reply:
This is not necessary, not having a value returned is not an problem: if we get one (i.e. that there wasn't any error) we use it to override the default service name, otherwise we stick with the default one (`fmt.Sprintf("mysql/replication/conn%d", i)`).